games/checkers/game.py

```diff
@@ -0,0 +1,83 @@
+from base.game import Game
+from games.checkers import EMPTY_SYMBOL
+from games.checkers.state import CheckersState
+from games.checkers.piece import BlackPiece, CheckersPiece, PlayerColor, WhitePiece
+from games.checkers.action import CheckersAction
+import numpy as np
+from typing import Sequence, Tuple
+from itertools import product
+
+
+class CheckersGame(Game[CheckersState, CheckersAction]):
+    """
+    The following rules are applied https://youtu.be/WD3NTNQElew with the exception, 
+    that the starting color is White instead of black.
+    """
+
+    def __init__(
+            self, board_shape: Tuple[int, int] = (8, 8),
+            player_color: PlayerColor = PlayerColor.WHITE, opponent_color: PlayerColor = PlayerColor.BLACK):
+        self.board_shape = board_shape
+        self._player_color, self._opponent_color = player_color, opponent_color
+        initial_state = self.initial_game_state()
+        super().__init__(initial_state)
+
+    def initial_game_state(self) -> CheckersState:
+        board = np.full(self.board_shape, EMPTY_SYMBOL, dtype=object)
+        for row, col in product(*[range(d) for d in self.board_shape]):
+            if col % 2 == ((row + 1) % 2):
+                if row < self.board_shape[0] / 2 - 1:
+                    piece = WhitePiece(row, col)
+                    board[row, col] = piece
+
+                elif row > self.board_shape[0] / 2:
+                    piece = BlackPiece(row, col)
+                    board[row, col] = piece
+
+        return CheckersState(board, self._player_color)
+
+    def actions_for(self, state: CheckersState, is_opponent: bool) -> Sequence[CheckersAction]:
+        actions = []
+        target_color = self._opponent_color if is_opponent else self._player_color
+        movable_pieces = [p for p in state.pieces if p.color == target_color]
+        for piece in movable_pieces:
+            for chain in self.find_beat_chains_for_piece(state.board, piece):
+                actions.append(CheckersAction(piece, chain, state))
+        jumps_exists = any(actions)
+        if jumps_exists:
+            return actions
```

Comment:
Przyjąłem wersję warcabów w której obowiązuje pobicie figury jeśli jest taka możliwość, więc w przypadku jeśli taka możliwość jest, to nie generuję dalej akcji.
Jeśli takiej możliwości nie ma, to generuję ruchy dla sąsiednich pozycji (chyba że są zajęte).